atlasdb-commons/src/test/java/com/palantir/common/concurrent/CoalescingSupplierTest.java

```diff
@@ -35,25 +35,57 @@
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
+import com.google.common.base.Throwables;
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.common.util.concurrent.ListeningExecutorService;
 import com.google.common.util.concurrent.MoreExecutors;
 import com.google.common.util.concurrent.Uninterruptibles;
 
+@RunWith(Parameterized.class)
 public class CoalescingSupplierTest {
     private static final int DEFAULT_VALUE = 123;
 
     private final Supplier<Integer> delegate = mock(Supplier.class);
     private final FreezableSupplier freezableDelegate = new FreezableSupplier(delegate);
-    private final CoalescingSupplier<Integer> supplier = new CoalescingSupplier<>(freezableDelegate);
+    private final CoalescingSupplier<Integer> coalescing = new CoalescingSupplier<>(freezableDelegate);
+
+    private final Supplier<Integer> supplier;
+
+    public CoalescingSupplierTest(String name, Object parameter) {
+        Function<CoalescingSupplierTest, Integer> factory = (Function) parameter;
+        supplier = () -> factory.apply(this);
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Object[] getParameters() {
+        return new Object[][] {
+                {"blocking", (Function<CoalescingSupplierTest, Integer>) test -> test.coalescing.get() },
+                {"async", (Function<CoalescingSupplierTest, Integer>) test -> unwrap(test.coalescing.getAsync()) }
+        };
+    }
+
+    private static <T> T unwrap(ListenableFuture<T> future) {
```

Comment:
Extract this to some utility class, as it's effectively duplicated from ConjureTimelockResource, and also exists in ALP